src/components/LHNOptionsList/OptionRowLHNData.js

```diff
@@ -88,6 +93,16 @@ function OptionRowLHNData({
     const parentReportAction = parentReportActions[fullReport.parentReportActionID];
 
     const optionItemRef = useRef();
+
+    const lastTransaction = useMemo(() => {
+        const sortedReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(reportActions);
+        const lastReportAction = _.first(sortedReportActions);
+        return TransactionUtils.getLinkedTransaction(lastReportAction);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [fullReport.reportID, receiptTransactions, reportActions]);
+
+    const memoizedLastTransaction = useDeepCompareMemo(lastTransaction);
```

Comment:
The `useDeepCompareMemo` does the deepEqal `isEqual` check. So even if the transaction has not changed but the reference is (you get a new object but with the same data), this will prevent rerendering

Reply:
If you always get the same object in situations where the transaction has not changed, then we dont have to use `useDeepCompareMemo`. We may remove it since `optionItem` memo does `deepEqual` check in `optionItem`. The only benefit of using `useDeepCompareMemo` is that this line won't have to be executed
`const item = SidebarUtils.getOptionData(fullReport, reportActions, personalDetails, preferredLocale, policy); 
![image](https://github.com/Expensify/App/assets/65986357/33085677-827e-45d4-b9b0-1ac19f3f451d)

`

Reply:
I can agree. And its not like you cant add these optimizations when there are performance issues. Would you also like me to remove the `useDeepCompareMemo` file which i created for this purpose?